flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java

```diff
@@ -259,6 +262,16 @@ private void abortCurrentProducer() {
         return currentProducer;
     }
 
+    @VisibleForTesting
+    KafkaRecordSerializationSchema<IN> getRecordSerializer() {
```

Comment:
Do you really need to expose the recordSerializer and the sink context? I only see them used by the `testNumRecordsOutErrorsCounterMetric` and I think both object are passed to writer during the construction so they are already accessible from the test class.

Reply:
Thanks for the hint.